source/violation.c

```diff
@@ -399,3 +407,189 @@ candl_violation_p candl_violation(osl_scop_p orig_scop,
   return violation;
 }
 
+static candl_violation_p violation_helper(osl_dependence_p orig_dependence,
+    osl_relation_p context, osl_statement_p source, osl_statement_p target,
+    int nb_parameters, candl_options_p options, PipOptions * pip_options) {
```

Comment:
Missing doxygen comment (the name of the parameter are transparent but there is still a lot).